packages/imperative/src/config/__tests__/Config.api.unit.test.ts

```diff
@@ -264,6 +265,72 @@ describe("Config API tests", () => {
+                const config = await Config.load(MY_APP);
+                const existsSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+                const readFileSpy = jest.spyOn(fs, "readFileSync");
+                const secureLoadSpy = jest.spyOn(config.api.secure, "loadCached");
+                const jsoncParseSpy = jest.spyOn(JSONC, "parse").mockImplementationOnce(() => { throw "failed to parse"; });
+                let caughtError: ImperativeError = {} as any;
+                try {
+                    config.api.layers.read();
+                } catch(err) {
+                    caughtError = err;
+                }
+                expect(existsSpy).toHaveBeenCalledTimes(5); // Once for each config layer and one more time for read
+                expect(readFileSpy).toHaveBeenCalledTimes(1);
+                expect(secureLoadSpy).toHaveBeenCalledTimes(0);
+                expect(jsoncParseSpy).toHaveBeenCalledTimes(1);
+                expect(caughtError.message).toContain("Please check this configuration file for errors");
+            });
+
+            it("should error when ignoreErrors is explicit false", async () => {
+                const config = await Config.load(MY_APP);
+                const existsSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+                const readFileSpy = jest.spyOn(fs, "readFileSync");
+                const secureLoadSpy = jest.spyOn(config.api.secure, "loadCached");
+                const jsoncParseSpy = jest.spyOn(JSONC, "parse").mockImplementationOnce(() => { throw "failed to parse"; });
+                let caughtError: ImperativeError = {} as any;
+                try {
+                    config.api.layers.read({user: false, global: false, ignoreErrors: false});
+                } catch(err) {
+                    caughtError = err;
+                }
+                expect(existsSpy).toHaveBeenCalledTimes(5); // Once for each config layer and one more time for read
+                expect(readFileSpy).toHaveBeenCalledTimes(1);
+                expect(secureLoadSpy).toHaveBeenCalledTimes(0);
+                expect(jsoncParseSpy).toHaveBeenCalledTimes(1);
+                expect(caughtError.message).toContain("Please check this configuration file for errors");
+            });
+
+            it("shouldnt error when ignoreErrors is true", async () => {
+                const config = await Config.load(MY_APP);
+                const existsSpy = jest.spyOn(fs, "existsSync").mockReturnValueOnce(true);
+                const readFileSpy = jest.spyOn(fs, "readFileSync");
+                const secureLoadSpy = jest.spyOn(config.api.secure, "loadCached");
+                const jsoncParseSpy = jest.spyOn(JSONC, "parse").mockImplementationOnce(() => { throw "failed to parse"; });
+
+                let logMsg: string = "Nothing logged";
+                jest.spyOn(Logger, "getImperativeLogger").mockImplementation(() => {
```

Comment:
We should mock the app logger here since that's what the Config class is using:
```suggestion
                jest.spyOn(Logger, "getAppLogger").mockImplementation(() => {
```